Minor improvements to OIDC docs (#558)
apache/pulsar-site: docs/security-openid-connect.md

The source code for the OpenID Connect implementation is in the [pulsar-broker-auth-oidc](https://github.com/apache/pulsar/blob/master/pulsar-broker-auth-oidc/) submodule in the Apache Pulsar git repo.

:::note

> Pulsar's OpenID Connect integration was introduced in Pulsar 3.0.0. As always, if you encounter any issues, please ask questions on Pulsar channels and open issues in GitHub.

Reviewer: ```suggestion
Pulsar's OpenID Connect integration is available from 3.0.0. 
```

> As always, if you encounter any issues, please ask questions on Pulsar channels and open issues in GitHub.

We do not need to add this because:
1) it's a common practice for developers
2) it's applicable for all features/improvements/... If we add this here, we need to add this everywhere

Author: > it's applicable for all features/improvements/... If we add this here, we need to add this everywhere

My intent is slightly different here than with other features. I agree that with open source projects, it is often implicit that we accept contributions from anyone interested in taking the time to give them. However, in this section, I wanted to convey that this is a new feature and there might be edge cases that are not covered in the docs. In some projects, they represent this by calling a feature "alpha", "beta", or "GA". We don't have those terms. Is there a different way I can word this to express that sentiment?

Reviewer: How about this?

::: note

Pulsar's OpenID Connect integration is available from 3.0.0. Currently, this feature is in beta.

:::

Author: After further thought, I think your first suggestion is correct.

7. When token validation is successful, the Pulsar Server extracts the `sub` claim from the token (or the configured `openIDRoleClaim`) and uses it as the principal for authorization.
8. When the token expires, the Pulsar Server challenges the client to re-authenticate with the Identity Provider and provide a new access token. If the client fails to re-authenticate, the Pulsar Server closes the connection.

## Enable OpenID Connect Authentication in the Broker and Proxy

> To configure Pulsar Servers to authenticate clients using OpenID Connect, add the following parameters to the `conf/broker.conf` and the `conf/proxy.conf`. If you use a standalone Pulsar, you need to add these parameters to the `conf/standalone.conf` file:

Reviewer: ```suggestion
To configure Pulsar servers to authenticate clients using OpenID Connect, add the following parameters to the `conf/broker.conf` and the `conf/proxy.conf`. If you use a standalone Pulsar, add these parameters to the `conf/standalone.conf` file:
```

Author: Do we capitalize broker and proxy in this context? I think I am going to replace server with broker and proxy.

Reviewer: I think we do not need to capitalize them because they are not proper nouns (e.g., Mac, Apple Watch).
